### ISSUE TITLE
LLVM_RUNTIMES_TARGET gets ignored while building libclc out-of-tree

### ISSUE THREAD

**Issue by slonkazoid** [llvm/llvm-project#191794]
I'm building libclc with the following command:

```sh
CC=$(llvm-config --bindir)/clang cmake libclc/CMakeLists.txt -G Ninja -DCMAKE_BUILD_TYPE=Release -DLLVM_DIR="$(llvm-config --cmakedir)" -DLLVM_RUNTIMES_TARGET=amdgcn-amd-amdhsa-llvm
```

but it ignores the option and builds for all runtimes present in the target llvm build anyways, and prints this warning:

```
CMake Warning:
  Manually-specified variables were not used by the project:

    LLVM_RUNTIMES_TARGET
```
